subiquity/ui/views/filesystem/partition.py

```diff
@@ -238,54 +213,99 @@ def __init__(self, model, controller, disk, partition=None):
-                Text(""),
-                button_pile([btn]),
                 ])
-        return body
+                focus_index = 2
+        rows.extend(self.form.as_rows())
+        widgets = [
+            Pile(rows),
+            Text(""),
+            self.form.buttons,
+        ]
+
+        if partition is None:
+            title = _("Adding partition to {}").format(disk.label)
+        else:
+            title = _("Editing partition {} of {}").format(
+                partition._number, disk.label)
 
-    def delete(self, sender):
-        self.controller.delete_partition(self.partition)
+        super().__init__(title, widgets, 0, focus_index)
+
+    def cancel(self, button=None):
+        self.parent.remove_overlay()
 
     def done(self, form):
         log.debug("Add Partition Result: {}".format(form.as_data()))
-        self.controller.partition_disk_handler(self.disk, self.partition,
-                                               form.as_data())
+        self.controller.partition_disk_handler(
+            self.disk, self.partition, form.as_data())
+        self.parent.refresh_model_inputs()
+        self.parent.remove_overlay()
 
 
-class FormatEntireView(PartitionFormatView):
+class FormatEntireStretchy(Stretchy):
 
-    def __init__(self, model, controller, volume, back):
-        self.model = model
-        self.controller = controller
-        self.volume = volume
-        if isinstance(volume, Disk):
-            self.title = _("Format and/or mount {}").format(volume.label)
-            self.footer = _("Format or mount whole disk.")
+    def __init__(self, parent, device):
+
+        self.device = device
+        self.model = parent.model
+        self.controller = parent.controller
+        self.parent = parent
+        mountpoint_to_devpath_mapping = (
+            self.model.get_mountpoint_to_devpath_mapping())
+        initial = {}
+        fs = device.fs()
+        if fs is not None:
+            initial['fstype'] = self.model.fs_by_name[fs.fstype]
+            mount = fs.mount()
+            if mount is not None:
+                initial['mount'] = mount.path
+                if mount.path in mountpoint_to_devpath_mapping:
+                    del mountpoint_to_devpath_mapping[mount.path]
         else:
-            self.title = _("Partition, format, "
-                           "and mount {}").format(volume.device.label)
-            self.footer = _("Edit partition details, format and mount.")
+            initial['fstype'] = self.model.fs_by_name[None]
+        self.form = PartitionForm(mountpoint_to_devpath_mapping, 0, initial)
+        self.form.remove_field('size')
 
-        super().__init__(None, volume, {}, back)
+        connect_signal(self.form, 'submit', self.done)
+        connect_signal(self.form, 'cancel', self.cancel)
+
+        rows = []
+        if isinstance(device, Disk):
+            rows = [
+                Text(_("Formatting and mounting a disk directly is unusual. "
```

Comment:
This seems like an odd comment/ui warning.

Reply:
Yeah. Currently the button for this functionality has the label "Format or create swap on entire device (unusual, advanced)" which is also pretty odd :)

It's a tricky one because:

1. It *is* a bit strange to mount a disk without partitioning
2. It is *not* strange to mount a RAID without partitioning
3. I can't think of a way to present the option in a way that makes it clear before choosing it that it's a bit of an odd thing to do.

Something about this reminds me that this branch probably doesn't handle formatting things as swap properly.